### ISSUE TITLE
Allow individual start nodes for native models

### ISSUE THREAD

**Issue by jackal2477** [xLightsSequencer/xLights#2334]
Some models work better as native models, but if they have an odd number of pixels, you have to change it to a custom model. For example, the Boscoyo Mesmerizer Spinner has 1117 pixels which is a prime number and does not allow a consistent number of pixels per strand. This can be worked around by converting to a custom model, but then it doesn't render as well as if it had been left as a native 'circle' model

**Comment by keithsw1111** [comment]
Why do you need to do this? I assume you are keen on doing this so you can upload the models. Have you looked into shadow models as a solution.

The challenge with this solution is it adds a lot of properties to all the models that we add it to. While it is not certain this is worse than shadow models which is to some degree a hack it is something to consider very carefully before we jump in and implement it. I do realise this is inconsistent with models like the matrix model but as we move away from start channels adding this seems like a regression in the simplification agenda. Maybe a better solution would be to allow you to specify the string length instead. It still has the issue of property growth but it is better in that it doesnt add more start channel settings.

**Comment by jackal2477** [comment]
I think that string length may be the right way to go. I have just started looking at shadow models, but as you say, it is kind of a kludge. As more and more vendors are coming out with high-density props, I think that this is going to come up more and more. Very few people are going to put 1k+ pixels on a single port, even if their controller supports it.

**Comment by AzGilrock** [comment]
Yeah but you only need this to support automatically configuring the controllers.  Everything can be setup manual like we did for years.  The more things that have been added to automate things the more confusion and problems that have been generated.

**Comment by jackal2477** [comment]
Actually I think the opposite is true (from a user not a programmer standpoint). When users have to use workarounds to use the automatic configuration, it causes confusion. Yes it can be done manually, but the automatic configuration is much easier and less confusing than dealing with overlapping channels. I think that the ultimate goal of auto-config is that the end user never even has to look at channels anymore

**Comment by AzGilrock** [comment]
Actually manual is pretty easy if you know what you're doing.  You can easily daisy chain all the models you want on the same controller and you never have to deal with overlaps.  The problem is the new people have no idea how we used to get things done so they are clueless as to what really goes on under the hood so when it doesn't work perfect it falls apart for them.  I have not seen the amount of support decrease due to adding the automatic controller configuration.  That's why it was added the developers were tired of spending hours trying to help people that can't figure anything out.  Well now its not really any better they've just built up a crew of people that sit in zoom and help people figure things out.

**Comment by cybercop23** [comment]
Fun day in the zoom room. We were working with a user several hours in the AM and then he came back in the afternoon with this exact prop. We showed him why/how to keep a native model and why it mattered - render differences.
We were trying to stay away from shadow models, he had a hard time grasping it, and were trying to only have the native model, 1117 pixels on 4 ports. That's 279.25/port, however the channel assigment was putting the start channel as 1 no matter what. We tried it with GE Grand Illussion at an even 1k and that worked as expected.
Given what we were tryting to do, although the number of pixels isn't even, can we just use the ceil(x) to get 1 .. n-1 and n be whatever is left? Heck... even if the last one had the same 280 pixles and this "wasting" 9 channels.., I think it will still be worth the ease of doing it vs a shadow model.

**Comment by merryoncherry** [comment]
I am curious what kind of model, as it's nice to think that custom models can do anything you'd want them to do.  All my spinners, for example, are custom models with submodels and they do everything regular spinners do plus some.

**Comment by computergeek1507** [comment]
With a Custom models you can make a submodel to emulate the default render style of the spinner. I personally don't understand why everyone makes shadow models when the default render can be done with a submodel. Custom models don't support "per strand" rendering styles which is the one downside. I want to add the ability to emulate per stand render styles with submodels someday.

**Comment by cybercop23** [comment]
@computergeek1507 I think the issue is with users buying and importing sequences. If they have to change each efect to match the original view, then it is not worth it.

@merryoncherry SS renders differently on the custom boscoyo vs custom. If one maps a PPD seq, it will be all f'ed.
We're not talking about sub-models, you can import, add/change SM on both types, it is the default render style that is different and hard to match to purchased/mapped sequences.

**Comment by computergeek1507** [comment]
> @computergeek1507 I think the issue is with users buying and importing sequences. If they have to change each efect to match the original view, then it is not worth it.
> 
> @merryoncherry SS renders differently on the custom boscoyo vs custom. If one maps a PPD seq, it will be all f'ed.
> We're not talking about sub-models, you can import, add/change SM on both types, it is the default render style that is different and hard to match to purchased/mapped sequences. 

You can make a submodel that matches the default render style and when you import use the submodels instead. Also can add it to your groups as the default too.

**Comment by merryoncherry** [comment]
All my spinners are custom models - hattitudes and such.  I put the appropriate submodel in the right groups instead of the main model and it works fine with all the import sequences, including a lot of PPD sequences.

**Comment by merryoncherry** [comment]
This topic came up 2 weeks ago here: https://auschristmaslighting.com/threads/spinner-effect.9082/post-131067

**Comment by cybercop23** [comment]
@merryoncherry agreed on custom models that can only be custom models, that's how I also do my 3 Quartets. For Mesmerizer or Grand Illusion that are just layered circles, you want to stay with native model and add submodules vs convert it to custom models. You can see the diff on the SS effect between the two, native acts like a spinner and native at model. Changing render styles doesn't work (at least for me).
We always recommend users keeping models as native models wherever possible.

**Comment by cybercop23** [comment]
#3148

**Comment by derwin12** [comment]
Closing as a duplicate and nothing to do in this ticket.
